src/encoding.ts

```diff
@@ -322,11 +322,11 @@ export function extractTransformsFromEncoding(oldEncoding: Encoding<Field>, conf
           // Add formatting to appropriate property based on the type of channel we're processing
           const format = getDateTimeComponents(timeUnit, config.axis.shortTimeLabels).join(' ');
           if (isNonPositionScaleChannel(channel)) {
-            newChannelDef['legend'] = {format, ...newChannelDef['legend']};
+            newChannelDef['legend'] = {format, formatType: 'time', ...newChannelDef['legend']};
           } else if (isPositionChannel) {
-            newChannelDef['axis'] = {format, ...newChannelDef['axis']};
+            newChannelDef['axis'] = {format, formatType: 'time', ...newChannelDef['axis']};
           } else if (channel === 'text' || channel === 'tooltip') {
-            newChannelDef['format'] = newChannelDef['format'] || format;
+            newChannelDef['format'] = {...(newChannelDef['format'] || format), ...{fotmatType: 'time'}};
```

Comment:
@kanit Would this else if statement be useful? When the channel is `text` or `tooltip`, wouldn't it go to the first if where `isNonPositionScaleChannel(channel)` is true?